python/tank/util/metrics.py

```diff
@@ -344,80 +417,67 @@ def data(self):
         return self._data
 
-
-class UserActivityMetric(ToolkitMetric):
-    """Convenience class for a user activity metric."""
-
-    def __init__(self, module, action):
-        """Initialize the metric with the module and action information.
-        
-        :param str module: Name of the module in which action was performed.
-        :param str action: The action that was performed.
-        
+    @classmethod
+    def log(cls, group, name, properties=None, log_once=False):
         """
-        super(UserActivityMetric, self).__init__({
-            "type": "user_activity",
-            "module": module,
-            "action": action,
-        })
-
-    def __repr__(self):
-        """Official str representation of the user activity metric."""
-        return "%s.%s" % (self._data["module"], self._data["action"])
-
-
-class UserAttributeMetric(ToolkitMetric):
-    """Convenience class for a user attribute metric."""
-
-    def __init__(self, attr_name, attr_value):
-        """Initialize the metric with the attribute name and value.
-        
-        :param str attr_name: Name of the attribute.
-        :param str attr_value: The value of the attribute.
+        Queue a Toolkit metric event with the given name for the given group on
+        the :class:`MetricsQueueSingleton` dispatch queue.
+
+        This method simply adds the metric event to the dispatch queue meaning that 
+        the metric has to be treated by a dispatcher to be posted.
+
+        :param str group: A group or category this metric event falls into.
+                          Although any values can be used, we encourage usage of the GROUP_*
+                          definitions above.
+        :param str name: A short descriptive event name or performed action, 
+                         e.g. 'Launched Command', 'Opened Workfile', etc..
+        :param dict properties: An optional dictionary of extra properties to be 
+                                attached to the metric event.
+        :param bool log_once: ``True`` if this metric should be ignored if it has
+                              already been logged. Defaults to ``False``.
 
         """
-        super(UserAttributeMetric, self).__init__({
-            "type": "user_attribute",
-            "attr_name": attr_name,
-            "attr_value": attr_value,
-        })
-
-    def __repr__(self):
-        """Official str representation of the user attribute metric."""
-        return "%s.%s" % (self._data["attr_name"], self._data["attr_value"])
+        log.debug("EventMetric.log('%s', '%s')" % (
+            str(cls(group, name, properties)),
+            str(log_once)
+        ))
+        MetricsQueueSingleton().log(
+            cls(group, name, properties),
+            log_once=log_once
+        )
 
 
 ###############################################################################
-# metrics logging convenience functions
+#
+# metrics logging convenience functions (All deprecated)
```

Comment:
i think these are all very internal and we don't expect anyone to call them directly so i suggest we delete them if not needed. @josh-t ?

Reply:
I was trying to play it safe just in case. I was thinking it wouldn't hurt us to leave them here for a bit as deprecated, then remove them after 6 months or so. But perhaps I'm overly cautious.

Reply:
IMHO the other deprecations make perfect sense. These ones i think are probably safe to remove straight away! Either way!